AssertionError in dsps_sed_demo

**Bug report**

It has been reported an AssertionError when running the smoke test for the dsps_sed_demo.ipynb python notebook.

AssertionError: /opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/rail/examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5 does not exist

The error occurs due to the not existence of the "ssp_data_fsps_v3.2_lgmet_age.h5" file. When not present in the default data directory, this file is curled from the web. In doing so, the resulting path looks wrong and the file is not found.

When comparing the dsps_sed_demo notebook with the one in rail_dsps, they had some differences. On my local machine, I matched the two versions and the dsps_sed_demo notebook in rail is running without any issue.

I'll create a branch where I will do the exact same modification to check whether the smoke test issue is fixed.

--- a/.github/workflows/smoke-test-head.yml
+++ b/.github/workflows/smoke-test-head.yml
@@ -41,7 +41,7 @@ jobs:
         echo "SPS_HOME=/opt/hostedtoolcache/Python/fsps" >> $GITHUB_ENV
     - name: Run notebooks
       run: |
-        rail render-nb --skip examples/creation_examples/plotting_interface_skysim_cosmoDC2_COSMOS2020_demo.ipynb examples/${{ matrix.notebook-dir }}_examples/*.ipynb
+        rail render-nb --skip examples/creation_examples/dsps_sed_demo.ipynb  --skip examples/creation_examples/plotting_interface_skysim_cosmoDC2_COSMOS2020_demo.ipynb examples/${{ matrix.notebook-dir }}_examples/*.ipynb
     - name: Send status to Slack app (RAIL CI Reporter)
       if: ${{ failure() && github.event_name != 'workflow_dispatch' }}
       id: slack

--- a/examples/creation_examples/dsps_sed_demo.ipynb
+++ b/examples/creation_examples/dsps_sed_demo.ipynb
@@ -11,13 +11,15 @@
    "source": [
     "# Demo of SED and photometry calculations with DSPS\n",
     "\n",
-    "author: Luca Tortorelli, Andrew Hearin\n",
+    "author: Luca Tortorelli\n",
     "\n",
-    "last run successfully: Aug 1, 2023\n",
+    "last run successfully: May 10, 2024\n",
     "\n",
-    "This notebook demonstrates some basic usage of the DSPS library. In particular, for a galaxy with some arbitrary star formation history, we'll see how to calculate its restframe SED, and its absolute and apparent magnitude in some band.\n",
+    "This notebook demonstrates some basic usage of the rail_dsps library. In particular, for a galaxy with some arbitrary star formation history, we'll see how to calculate its restframe SED, and its absolute and apparent magnitude in some band.\n",
     "\n",
-    "SPS calculations are based on a set of template SEDs of simple stellar populations (SSPs). Supplying such templates is outside the planned scope of the DSPS package, and so they will need to be retrieved from some other library. For example, the FSPS library supplies such templates in a convenient form."
+    "SPS calculations are based on a set of template SEDs of simple stellar populations (SSPs). Supplying such templates is outside the planned scope of the DSPS package, and so they will need to be retrieved from some other library. For example, the FSPS library supplies such templates in a convenient form.\n",
+    "\n",
+    "In case the user is not aware of how to generate these templates, the code automatically downloads a pre-defined library from NERSC."
    ]
   },
   {
@@ -41,7 +43,7 @@
     }
    },
    "source": [
-    "The SingleSedModeler class allows the user to generate a single rest-frame SED at the time with DSPS.\n",
+    "The SingleSedModeler class allows the user to generate a single rest-frame SED at the time with rail_dsps.\n",
     "\n",
     "Some example galaxy properties that are required to create a single SED model are generated via this notebook and stored into an hdf5 table. The required galaxy properties are:\n",
     "\n",
@@ -54,22 +56,26 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
    "metadata": {
     "collapsed": false,
     "jupyter": {
      "outputs_hidden": false
+    },
+    "ExecuteTime": {
+     "end_time": "2024-05-10T15:41:08.510620Z",
+     "start_time": "2024-05-10T15:41:05.620759Z"
     }
    },
-   "outputs": [],
    "source": [
     "import os\n",
     "import rail.dsps\n",
     "from rail.core.stage import RailStage\n",
-    "from rail.core.data import TableHandle\n",
+    "from rail.core.data import Hdf5Handle\n",
     "import numpy as np\n",
     "import h5py"
-   ]
+   ],
+   "outputs": [],
+   "execution_count": 1
   },
   {
    "cell_type": "code",
@@ -85,10 +91,9 @@
     "DS = RailStage.data_store\n",
     "DS.__class__.allow_overwrite = True\n",
     "\n",
-    "RAIL_DSPS_DIR = os.path.abspath(os.path.join(os.path.dirname(rail.dsps.__file__), '..', '..'))\n",
-    "default_rail_fsps_files_folder = os.path.join(RAIL_DSPS_DIR, 'rail', 'examples_data', 'creation_data',\n",
-    "                                              'data', 'dsps_default_data')\n",
-    "input_file = os.path.join(default_rail_fsps_files_folder, 'input_galaxy_properties_dsps.hdf5')"
+    "RAIL_DSPS_DIR = os.path.abspath(os.path.join(os.path.dirname(rail.dsps.__file__), '..'))\n",
+    "default_rail_fsps_files_folder = os.path.join(RAIL_DSPS_DIR, 'examples_data/creation_data/data/dsps_default_data')\n",
+    "trainFile = os.path.join(default_rail_fsps_files_folder, 'input_galaxy_properties_dsps.hdf5')"
    ]
   },
   {
@@ -117,7 +122,7 @@
     "stellar_metallicity = np.full(n_galaxies, gal_lgmet)\n",
     "stellar_metallicity_scatter = np.full(n_galaxies, gal_lgmet_scatter)\n",
     "\n",
-    "with h5py.File(input_file, 'w') as h5table:\n",
+    "with h5py.File(trainFile, 'w') as h5table:\n",
     "    h5table.create_dataset(name='redshifts', data=redshift)\n",
     "    h5table.create_dataset(name='cosmic_time_grid', data=cosmic_time_grid)\n",
     "    h5table.create_dataset(name='star_formation_history', data=star_formation_history)\n",
@@ -135,10 +140,7 @@
     }
    },
    "outputs": [],
-   "source": [
-    "trainFile = os.path.join(input_file)\n",
-    "training_data = DS.read_file(\"training_data\", TableHandle, trainFile)"
-   ]
+   "source": "training_data = DS.read_file(\"training_data\", Hdf5Handle, trainFile)"
   },
   {
    "cell_type": "markdown",
@@ -164,13 +166,14 @@
    "outputs": [],
    "source": [
     "dspssinglesedmodeler = rail.dsps.DSPSSingleSedModeler.make_stage(name='DSPSSingleSedModeler',\n",
-    "                                                                 ssp_templates_file=os.path.join(RAIL_DSPS_DIR,'rail/examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
+    "                                                                 ssp_templates_file=os.path.join(RAIL_DSPS_DIR,'examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
     "                                                                 redshift_key='redshifts',\n",
     "                                                                 cosmic_time_grid_key='cosmic_time_grid',\n",
     "                                                                 star_formation_history_key='star_formation_history',\n",
     "                                                                 stellar_metallicity_key='stellar_metallicity',\n",
     "                                                                 stellar_metallicity_scatter_key='stellar_metallicity_scatter',\n",
-    "                                                                 restframe_sed_key='restframe_seds', default_cosmology=True)"
+    "                                                                 restframe_sed_key='restframe_seds', default_cosmology=True,\n",
+    "                                                                 min_wavelength=250, max_wavelength=12000)"
    ]
   },
   {
@@ -183,9 +186,7 @@
     }
    },
    "outputs": [],
-   "source": [
-    "dspssinglesedmodel = dspssinglesedmodeler.fit_model(input_data=training_data)"
-   ]
+   "source": "dspssinglesedmodel = dspssinglesedmodeler.fit_model(input_data=training_data)"
   },
   {
    "cell_type": "code",
@@ -239,7 +240,7 @@
     "import os\n",
     "import rail.dsps\n",
     "from rail.core.stage import RailStage\n",
-    "from rail.core.data import TableHandle\n",
+    "from rail.core.data import Hdf5Handle\n",
     "import numpy as np\n",
     "import h5py"
    ]
@@ -258,9 +259,8 @@
     "DS = RailStage.data_store\n",
     "DS.__class__.allow_overwrite = True\n",
     "\n",
-    "default_rail_fsps_files_folder = os.path.join(RAIL_DSPS_DIR, 'rail', 'examples_data', 'creation_data',\n",
-    "                                              'data', 'dsps_default_data')\n",
-    "input_file = os.path.join(default_rail_fsps_files_folder, 'input_galaxy_properties_dsps.hdf5')"
+    "default_rail_fsps_files_folder = os.path.join(RAIL_DSPS_DIR, 'examples_data/creation_data/data/dsps_default_data')\n",
+    "trainFile = os.path.join(default_rail_fsps_files_folder, 'input_galaxy_properties_dsps.hdf5')"
    ]
   },
   {
@@ -289,7 +289,7 @@
     "stellar_metallicity = np.full(n_galaxies, gal_lgmet)\n",
     "stellar_metallicity_scatter = np.full(n_galaxies, gal_lgmet_scatter)\n",
     "\n",
-    "with h5py.File(input_file, 'w') as h5table:\n",
+    "with h5py.File(trainFile, 'w') as h5table:\n",
     "    h5table.create_dataset(name='redshifts', data=redshift)\n",
     "    h5table.create_dataset(name='cosmic_time_grid', data=cosmic_time_grid)\n",
     "    h5table.create_dataset(name='star_formation_history', data=star_formation_history)\n",
@@ -307,10 +307,7 @@
     }
    },
    "outputs": [],
-   "source": [
-    "trainFile = os.path.join(input_file)\n",
-    "training_data = DS.read_file(\"training_data\", TableHandle, trainFile)"
-   ]
+   "source": "training_data = DS.read_file(\"training_data\", Hdf5Handle, trainFile)"
   },
   {
    "cell_type": "code",
@@ -325,13 +322,14 @@
    "source": [
     "dspspopulationsedmodeler = rail.dsps.DSPSPopulationSedModeler.make_stage(name='DSPSPopulationSedModeler',\n",
     "                                                                         ssp_templates_file=os.path.join(RAIL_DSPS_DIR,\n",
-    "                                                                         'rail/examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
+    "                                                                         'examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
     "                                                                         redshift_key='redshifts',\n",
     "                                                                         cosmic_time_grid_key='cosmic_time_grid',\n",
     "                                                                         star_formation_history_key='star_formation_history',\n",
     "                                                                         stellar_metallicity_key='stellar_metallicity',\n",
     "                                                                         stellar_metallicity_scatter_key='stellar_metallicity_scatter',\n",
-    "                                                                         restframe_sed_key='restframe_seds', default_cosmology=True)"
+    "                                                                         restframe_sed_key='restframe_seds', default_cosmology=True,\n",
+    "                                                                         min_wavelength=250, max_wavelength=12000)"
    ]
   },
   {
@@ -391,6 +389,7 @@
     "- the absolute and apparent magnitudes dataset keyword of the output hdf5 table\n",
     "- the folder path containing the filter bands\n",
     "- the name of the filter bands in order of increasing wavelength\n",
+    "- the minimum and maximum wavelength used in the DSPSPopulationSedModeler class to generate the rest-frame SEDs\n",
     "- the path to the SSP template files\n",
     "- a boolean keyword to use (True) the default cosmology in DSPS.\n",
     "\n",
@@ -411,7 +410,7 @@
     "import os\n",
     "import rail.dsps\n",
     "from rail.core.stage import RailStage\n",
-    "from rail.core.data import TableHandle"
+    "from rail.core.data import Hdf5Handle"
    ]
   },
   {
@@ -428,7 +427,7 @@
     "DS = RailStage.data_store\n",
     "DS.__class__.allow_overwrite = True\n",
     "\n",
-    "input_file = 'model_DSPSPopulationSedModeler.hdf5'"
+    "trainFile = 'model_DSPSPopulationSedModeler.hdf5'"
    ]
   },
   {
@@ -441,10 +440,7 @@
     }
    },
    "outputs": [],
-   "source": [
-    "trainFile = os.path.join(input_file)\n",
-    "training_data = DS.read_file(\"training_data\", TableHandle, trainFile)"
-   ]
+   "source": "training_data = DS.read_file(\"training_data\", Hdf5Handle, trainFile)"
   },
   {
    "cell_type": "code",
@@ -463,11 +459,12 @@
     "                                                         absolute_mags_key='rest_frame_absolute_mags',\n",
     "                                                         apparent_mags_key='apparent_mags',\n",
     "                                                         filter_folder=os.path.join(RAIL_DSPS_DIR,\n",
-    "                                                         'rail/examples_data/creation_data/data/dsps_default_data/filters'),\n",
+    "                                                         'examples_data/creation_data/data/dsps_default_data/filters'),\n",
     "                                                         instrument_name='lsst',\n",
     "                                                         wavebands='u,g,r,i,z,y',\n",
+    "                                                         min_wavelength=250, max_wavelength=12000,\n",
     "                                                         ssp_templates_file=os.path.join(RAIL_DSPS_DIR,\n",
-    "                                                         'rail/examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
+    "                                                         'examples_data/creation_data/data/dsps_default_data/ssp_data_fsps_v3.2_lgmet_age.h5'),\n",
     "                                                         default_cosmology=True)"
    ]
   },

--- a/examples/creation_examples/plotting_interface_skysim_cosmoDC2_COSMOS2020_demo.ipynb
+++ b/examples/creation_examples/plotting_interface_skysim_cosmoDC2_COSMOS2020_demo.ipynb
@@ -9,7 +9,6 @@
    },
    "outputs": [],
    "source": [
-    "import GCRCatalogs\n",
     "import matplotlib.pyplot as plt\n",
     "import pandas as pd\n",
     "import corner\n",
@@ -23,9 +22,7 @@
    "cell_type": "markdown",
    "id": "6b84d8ef-ea56-4bb8-b15f-5a5b10691d66",
    "metadata": {},
-   "source": [
-    "## Read skysim v3.1.0 and cosmoDC2 v1.1.4 catalogs with the GCRCatalogs class"
-   ]
+   "source": "## Read skysim v3.1.0 and cosmoDC2 v1.1.4 catalogs with the GCRCatalogs class, alternatively curl from web repository"
   },
   {
    "cell_type": "code",
@@ -36,103 +33,41 @@
    },
    "outputs": [],
    "source": [
-    "skysim3 = GCRCatalogs.load_catalog('skysim_v3.1.0')\n",
-    "cosmodc2 = GCRCatalogs.load_catalog('cosmoDC2_v1.1.4_small')"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "a0675c78-2fff-4692-bfbb-de25a5a60e62",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "skysim3_allcols = skysim3.list_all_native_quantities()\n",
-    "cosmodc2_allcols = cosmodc2.list_all_native_quantities()"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "c12c1711-b8a7-4d60-9d94-4455b37a2490",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "skysim3_relevantcols = ['LSST_obs_g','LSST_obs_i','LSST_obs_r','LSST_obs_u','LSST_obs_y','LSST_obs_z','LSST_rest_G','LSST_rest_I','LSST_rest_R',\n",
+    "try:\n",
+    "    import GCRCatalogs\n",
+    "    skysim3 = GCRCatalogs.load_catalog('skysim_v3.1.0')\n",
+    "    cosmodc2 = GCRCatalogs.load_catalog('cosmoDC2_v1.1.4_small')\n",
+    "    skysim3_allcols = skysim3.list_all_native_quantities()\n",
+    "    cosmodc2_allcols = cosmodc2.list_all_native_quantities()\n",
+    "    skysim3_relevantcols = ['LSST_obs_g','LSST_obs_i','LSST_obs_r','LSST_obs_u','LSST_obs_y','LSST_obs_z','LSST_rest_G','LSST_rest_I','LSST_rest_R',\n",
     "                        'LSST_rest_U' ,'LSST_rest_Y','LSST_rest_Z','lg_met_mean','lg_met_scatter','log_sm','log_ssfr','obs_sfr','obs_sm',\n",
     "                        'redshift','sfr','HSC_obs_g','HSC_obs_i','HSC_obs_r','HSC_obs_y','HSC_obs_z','HSC_rest_G','HSC_rest_I','HSC_rest_R',\n",
     "                        'HSC_rest_Y','HSC_rest_Z', 'diffmah_early_index', 'lg_met_scatter', \n",
     "                        'diffstar_u_indx_lo', 'diffstar_u_tau_dep', 'diffmah_mah_logtc', 'diffmah_logmp_fit',\n",
     "                        'lg_met_mean', 'diffstar_u_qs', 'diffstar_u_q_rejuv', 'diffstar_u_lgmcrit',\n",
     "                        'diffstar_u_qt', 'diffmah_late_index', 'diffstar_u_q_drop', 'diffstar_u_lgy_at_mcrit',\n",
-    "                        'diffstar_u_indx_hi']"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "f7f7ed74-1146-4094-b95a-3524e208bb49",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "cosmodc2_relevantcols = ['LSST_filters/magnitude:LSST_u:rest','LSST_filters/magnitude:LSST_g:rest','LSST_filters/magnitude:LSST_r:rest',\n",
+    "                        'diffstar_u_indx_hi']\n",
+    "    cosmodc2_relevantcols = ['LSST_filters/magnitude:LSST_u:rest','LSST_filters/magnitude:LSST_g:rest','LSST_filters/magnitude:LSST_r:rest',\n",
     "                         'LSST_filters/magnitude:LSST_i:rest','LSST_filters/magnitude:LSST_z:rest','LSST_filters/magnitude:LSST_y:rest',\n",
     "                         'LSST_filters/magnitude:LSST_u:observed','LSST_filters/magnitude:LSST_g:observed','LSST_filters/magnitude:LSST_r:observed',\n",
     "                         'LSST_filters/magnitude:LSST_i:observed','LSST_filters/magnitude:LSST_z:observed','LSST_filters/magnitude:LSST_y:observed',\n",
-    "                         'baseDC2/sfr','baseDC2/redshift','baseDC2/obs_sfr','baseDC2/obs_sm','baseDC2/sm']"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "f8769596-f16f-46f5-84ed-e88aa56f0ef0",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "skysim3_quantities = skysim3.get_quantities(skysim3_relevantcols, native_filters=['healpix_pixel == 9816'])\n",
-    "cosmodc2_quantities = cosmodc2.get_quantities(cosmodc2_relevantcols, native_filters=['healpix_pixel == 9816'])"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "20db1f1b-d877-4134-9991-7b448d09a584",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "skysim3_df = pd.DataFrame(skysim3_quantities)\n",
-    "cosmodc2_df = pd.DataFrame(cosmodc2_quantities)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "cfdc79d5-b9d7-4b3e-9ae0-b7ebcac8838f",
-   "metadata": {},
-   "source": [
-    "## Select a random subset of 100k objects per catalog to facilitate plotting"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "934e777e-232e-4bc8-a59e-0331541aa074",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "skysim3_df_100k = skysim3_df.sample(n=100000)\n",
-    "cosmodc2_df_100k = cosmodc2_df.sample(n=100000)"
+    "                         'baseDC2/sfr','baseDC2/redshift','baseDC2/obs_sfr','baseDC2/obs_sm','baseDC2/sm']\n",
+    "    skysim3_quantities = skysim3.get_quantities(skysim3_relevantcols, native_filters=['healpix_pixel == 9816'])\n",
+    "    cosmodc2_quantities = cosmodc2.get_quantities(cosmodc2_relevantcols, native_filters=['healpix_pixel == 9816'])\n",
+    "    skysim3_df = pd.DataFrame(skysim3_quantities)\n",
+    "    cosmodc2_df = pd.DataFrame(cosmodc2_quantities)\n",
+    "    # Select a random subset of 100k objects per catalog to facilitate plotting\n",
+    "    skysim3_df_100k = skysim3_df.sample(n=100000)\n",
+    "    cosmodc2_df_100k = cosmodc2_df.sample(n=100000)\n",
+    "    \n",
+    "except ModuleNotFoundError:\n",
+    "    output_directory = os.getcwd()\n",
+    "    os.system('curl -O https://portal.nersc.gov/cfs/lsst/PZ/skysim_v3.1.0_100kpublic.csv '\n",
+    "              '--output-dir {}'.format(output_directory))\n",
+    "    os.system('curl -O https://portal.nersc.gov/cfs/lsst/PZ/cosmodc2_100kpublic.csv '\n",
+    "              '--output-dir {}'.format(output_directory))\n",
+    "    skysim3_df_100k = pd.read_csv(os.path.join(output_directory, 'skysim_v3.1.0_100kpublic.csv'))\n",
+    "    cosmodc2_df_100k = pd.read_csv(os.path.join(output_directory, 'cosmodc2_100kpublic.csv'))"
    ]
   },
   {
@@ -757,43 +692,11 @@
    "source": [
     "output_directory = os.getcwd()\n",
     "\n",
-    "if not os.path.isfile(os.path.join(output_directory,'COSMOS2020_CLASSIC_R1_v2.0.fits')):\n",
-    "    os.system('curl -O https://portal.nersc.gov/cfs/lsst/PZ/COSMOS2020_CLASSIC_R1_v2.0.fits '\n",
+    "if not os.path.isfile(os.path.join(output_directory,'COSMOS2020_CLASSIC_R1_v2.0_100k.fits')):\n",
+    "    os.system('curl -O https://portal.nersc.gov/cfs/lsst/PZ/COSMOS2020_CLASSIC_R1_v2.0_100k.fits '\n",
     "              '--output-dir {}'.format(output_directory))\n",
     "\n",
-    "cosmos2020 = Table.read(os.path.join(output_directory,'COSMOS2020_CLASSIC_R1_v2.0.fits'), format='fits')"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "30624b69-b012-4f3c-994e-03f2fb0e1352",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "mask_bad_values = np.where((cosmos2020['CFHT_u_MAG_AUTO'] > 0) & \n",
-    "                             (cosmos2020['HSC_i_MAG_AUTO'] > 0)&\n",
-    "                             (cosmos2020['HSC_g_MAG_AUTO'] > 0)&\n",
-    "                             (cosmos2020['HSC_r_MAG_AUTO'] > 0) & \n",
-    "                             (cosmos2020['HSC_z_MAG_AUTO'] > 0)&\n",
-    "                             (cosmos2020['HSC_y_MAG_AUTO'] > 0)&\n",
-    "                             (cosmos2020['UVISTA_J_MAG_AUTO'] > 0) &\n",
-    "                             (cosmos2020['UVISTA_H_MAG_AUTO'] > 0) & \n",
-    "                             (cosmos2020['UVISTA_Ks_MAG_AUTO'] > 0))"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "e4dd14a6-585c-45bf-9522-6454ef335f6b",
-   "metadata": {
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "cosmos2020 = cosmos2020[mask_bad_values]"
+    "cosmos2020 = Table.read(os.path.join(output_directory,'COSMOS2020_CLASSIC_R1_v2.0_100k.fits'), format='fits')"
    ]
   },
   {
@@ -813,8 +716,8 @@
    },
    "outputs": [],
    "source": [
-    "skysim3_df_lsst_y1_all = skysim3_df.loc[skysim3_df['LSST_obs_i']<=24.1]\n",
-    "cosmodc2_df_lsst_y1_all = cosmodc2_df.loc[cosmodc2_df['LSST_filters/magnitude:LSST_i:observed']<=24.1]"
+    "skysim3_df_lsst_y1 = skysim3_df_100k.loc[skysim3_df_100k['LSST_obs_i']<=24.1]\n",
+    "cosmodc2_df_lsst_y1 = cosmodc2_df_100k.loc[cosmodc2_df_100k['LSST_filters/magnitude:LSST_i:observed']<=24.1]"
    ]
   },
   {
@@ -841,8 +744,8 @@
    },
    "outputs": [],
    "source": [
-    "skysim3_df_lsst_y1 = skysim3_df_lsst_y1_all.sample(n=100000)\n",
-    "cosmodc2_df_lsst_y1 = cosmodc2_df_lsst_y1_all.sample(n=100000)"
+    "#skysim3_df_lsst_y1 = skysim3_df_lsst_y1_all.sample(n=100000)\n",
+    "#cosmodc2_df_lsst_y1 = cosmodc2_df_lsst_y1_all.sample(n=100000)"
    ]
   },
   {
@@ -1496,13 +1399,14 @@
     "plt.rcParams['ytick.labelsize']=15\n",
     "\n",
     "extents = [(-1.5,4.), (-1.5,4.), (-1.5,4.), (-1.5,4.),(-1.5,4.)]\n",
+    "colours = [r'$(u-g)_{obs}$',r'$(g-r)_{obs}$',r'$(r-i)_{obs}$',r'$(i-z)_{obs}$',r'$(z-y)_{obs}$']\n",
     "\n",
     "fig = corner.corner(cosmos2020_colors[:,:5], hist_kwargs=dict(density=True, linewidth=1.5), range=extents,\n",
-    "                    color='blue', labels=colours[:5], label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
+    "                    color='blue', labels=colours, label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
     "                    plot_datapoints=False, levels=(1-np.exp(-0.5),1-np.exp(-2),1-np.exp(-4.5)), show_titles=False,\n",
     "                    title_fmt='.4f')\n",
     "fig = corner.corner(dsps_colors[:,:5], fig=fig, hist_kwargs=dict(density=True, linewidth=1.5), range=extents,\n",
-    "                    color='red', labels=colours[:5], label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
+    "                    color='red', labels=colours, label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
     "                    plot_datapoints=False, levels=(1-np.exp(-0.5),1-np.exp(-2),1-np.exp(-4.5)), show_titles=False,\n",
     "                    title_fmt='.4f')\n",
     "                    \n",
@@ -1530,13 +1434,14 @@
     "plt.rcParams['ytick.labelsize']=15\n",
     "\n",
     "extents = [(-1.5,4.), (-1.5,4.), (-1.5,4.)]\n",
+    "colours = [r'$(y-J)_{obs}$',r'$(J-H)_{obs}$',r'$(H-Ks)_{obs}$']\n",
     "\n",
     "fig = corner.corner(cosmos2020_colors[:,5:], hist_kwargs=dict(density=True, linewidth=1.5), range=extents,\n",
-    "                    color='blue', labels=colours[5:], label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
+    "                    color='blue', labels=colours, label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
     "                    plot_datapoints=False, levels=(1-np.exp(-0.5),1-np.exp(-2),1-np.exp(-4.5)), show_titles=False,\n",
     "                    title_fmt='.4f')\n",
     "fig = corner.corner(dsps_colors[:,5:], fig=fig, hist_kwargs=dict(density=True, linewidth=1.5), range=extents,\n",
-    "                    color='red', labels=colours[5:], label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
+    "                    color='red', labels=colours, label_kwargs={\"fontsize\":30}, use_math_text=True,\n",
     "                    plot_datapoints=False, levels=(1-np.exp(-0.5),1-np.exp(-2),1-np.exp(-4.5)), show_titles=False,\n",
     "                    title_fmt='.4f')\n",
     "                    \n",

